ee/clickhouse/queries/trends/lifecycle.py

```diff
@@ -1,86 +1,41 @@
+# during that period and their creation dates.
 
-class ClickhouseLifecycle(LifecycleTrend):
-    def get_interval(self, interval: str) -> Tuple[Union[timedelta, relativedelta], str, str]:
-        if interval == "hour":
-            return timedelta(hours=1), "1 HOUR", "1 MINUTE"
-        elif interval == "day":
-            return timedelta(days=1), "1 DAY", "1 HOUR"
-        elif interval == "week":
-            return timedelta(weeks=1), "1 WEEK", "1 DAY"
-        elif interval == "month":
-            return relativedelta(months=1), "1 MONTH", "1 DAY"
-        else:
-            raise ValidationError("{interval} not supported")
 
+class ClickhouseLifecycle(LifecycleTrend):
     def _format_lifecycle_query(self, entity: Entity, filter: Filter, team_id: int) -> Tuple[str, Dict, Callable]:
-        date_from = filter.date_from
-
-        if not date_from:
-            date_from = get_earliest_timestamp(team_id)
-
-        interval = filter.interval
-        num_intervals, seconds_in_interval, _ = get_time_diff(interval, filter.date_from, filter.date_to, team_id)
-        interval_increment, interval_string, sub_interval_string = self.get_interval(interval)
-        trunc_func = get_trunc_func_ch(interval)
-        event_query = ""
-        event_params: Dict[str, Any] = {}
-
-        props_to_filter = [*filter.properties, *entity.properties]
-        prop_filters, prop_filter_params = parse_prop_clauses(props_to_filter, group_properties_joined=False)
-
-        _, _, date_params = parse_timestamps(filter=filter, team_id=team_id)
-
-        if entity.type == TREND_FILTER_TYPE_ACTIONS:
-            try:
-                action = entity.get_action()
-                event_query, event_params = format_action_filter(action)
-            except:
-                return "", {}, self._parse_result(filter, entity)
-        else:
-            event_query = "event = %(event)s"
-            event_params = {"event": entity.id}
+        event_query, event_params = LifecycleEventQuery(team_id=team_id, filter=filter).get_query()
 
         return (
-            LIFECYCLE_SQL.format(
-                interval=interval_string,
-                trunc_func=trunc_func,
-                event_query=event_query,
-                filters=prop_filters,
-                sub_interval=sub_interval_string,
-                GET_TEAM_PERSON_DISTINCT_IDS=get_team_distinct_ids_query(team_id),
-            ),
-            {
-                "team_id": team_id,
-                "prev_date_from": (date_from - interval_increment).strftime(
-                    "%Y-%m-%d{}".format(" %H:%M:%S" if filter.interval == "hour" else " 00:00:00")
-                ),
-                "num_intervals": num_intervals,
-                "seconds_in_interval": seconds_in_interval,
-                **event_params,
-                **date_params,
-                **prop_filter_params,
-            },
+            LIFECYCLE_SQL.format(events_query=event_query, interval_expr=filter.interval),
```

Comment:
Can we assume that `filter.interval` has been validated and therefore is safe to include in the format?

Reply:
Yep. `def interval` has validation built in already :)